src/views/SignUp.js

```diff
@@ -0,0 +1,175 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import {
+  handleRegistration,
+  handleInputChange,
+  handleBlur
+} from "../redux/actions/registrationActions";
+import TextInput from "../components/common/Inputs/TextInput";
+import checkField from "../utils/checkField";
+import FormButton from "../components/common/Buttons/FormButton";
+
+export class SignUp extends Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+  }
+
+  handleChange(event) {
+    event.stopPropagation();
+    const { name, value } = event.target;
+    const { dispatch } = this.props;
+    dispatch(handleInputChange(name, value));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { dispatch, user, message } = this.props;
+    if (!message) {
+      dispatch(handleRegistration(user));
+    }
+  }
+
+  handleBlur(event) {
+    event.preventDefault();
+    const { name, value } = event.target;
+    const { dispatch } = this.props;
+    dispatch(handleBlur(name, value));
+  }
+
+  render() {
+    const { message, user, submitted, success, history } = this.props;
+    if (success) {
+      history.push("/profile");
+    }
+    return (
+      <div className="auth">
+        <div className="row">
+          <div className="col-md-6 hide-sm">
+            <div className="auth-left">
+              <div className="logo">
+                <img
```

Comment:
You can use Logo component for this line.

Reply:
done